src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts

```diff
@@ -0,0 +1,127 @@
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { Component } from '@angular/core';
+import {
+  FeedbackMcqQuestionDetails,
+  FeedbackParticipantType,
+  FeedbackQuestionType,
+} from '../../../../types/api-output';
+import { QuestionEditDetailsFormComponent } from './question-edit-details-form.component';
+
+/**
+ * Question details edit form component for mcq scale question.
+ */
+@Component({
+  selector: 'tm-mcq-question-edit-details-form',
+  templateUrl: './mcq-question-edit-details-form.component.html',
+  styleUrls: ['./mcq-question-edit-details-form.component.scss'],
+})
+export class McqQuestionEditDetailsFormComponent
+    extends QuestionEditDetailsFormComponent<FeedbackMcqQuestionDetails> {
+
+  readonly PARTICIPANT_TYPES: string[] = [FeedbackParticipantType.STUDENTS,
+    FeedbackParticipantType.STUDENTS_EXCLUDING_SELF, FeedbackParticipantType.TEAMS,
+    FeedbackParticipantType.TEAMS_EXCLUDING_SELF, FeedbackParticipantType.INSTRUCTORS];
+
+  constructor() {
+    super({
+      hasAssignedWeights: false,
+      mcqWeights: [],
+      mcqOtherWeight: 0,
+      numOfMcqChoices: 2,
+      mcqChoices: [' ', ' '],
+      otherEnabled: false,
+      generatedOptionsEnabled: false,
+      generateOptionsFor: FeedbackParticipantType.NONE,
+      questionText: '',
+      questionType: FeedbackQuestionType.MCQ,
+    });
+  }
+
+  /**
+   * Reorders the list on dragging the Mcqs.
+   */
+  onMcqDropped(event: CdkDragDrop<string[]>): void {
+    moveItemInArray(this.model.mcqChoices, event.previousIndex, event.currentIndex);
+    moveItemInArray(this.model.mcqWeights, event.previousIndex, event.currentIndex);
```

Comment:
What if there is no weight?

Reply:
Seems to work fine even when hasAssignedWeights is false. The moveItemInArray implementation seems to take care of empty array cases